youtube_dl/extractor/openload.py

```diff
@@ -75,17 +75,20 @@ def _real_extract(self, url):
             '<span[^>]+id="[^"]+"[^>]*>([0-9]+)</span>',
             webpage, 'openload ID')
 
-        first_three_chars = int(float(ol_id[0:][:3]))
-        fifth_char = int(float(ol_id[3:5]))
-        urlcode = ''
-        num = 5
+        first_two_chars = int(float(ol_id[0:][:2]))
+        urlcode = {}
+        num = 2
 
         while num < len(ol_id):
-            urlcode += compat_chr(int(float(ol_id[num:][:3])) +
-                                  first_three_chars - fifth_char * int(float(ol_id[num + 3:][:2])))
+            key = int(float(ol_id[num + 3:][:2]))
+            urlcode[key] = compat_chr(int(float(ol_id[num:][:3])) - first_two_chars)
             num += 5
+        
+        sorted(urlcode, key=lambda key: urlcode[key])  
```

Comment:
Probably this will work instead?

`urllink = ''.join([str(urlcode[key]) for key in sorted(urlcode.keys())])`